[17.0][ADD] hr_attendance_overtime_negative_hours

xaviedoanhduy opened the pull request:
- this PR add a new module to branch 17.0: `hr_leave_dashboard_extra_hours`

- it is based upon the migration of a 15.0 module `hr_attendance_overtime_negative_hours` .

- in v15, the module made sense because if the employee had some Negative Hours, the Extra Hours card would not appear in the Time Off dashboard

-  in v17, a module is still needed to ensure the Extra Hours card would appear, but for slightly different reasons: this Extra Hours card will never appear anyway because the Extra Hours leave type has `requires_allocation == no`  [more detail in here](https://github.com/odoo/odoo/blob/ea4983e77f65d8d963818bb92f95f998db2f1846/addons/hr_holidays/models/hr_leave_type.py#L417)

- we renamed the module from `hr_attendance_overtime_negative_hours` to `hr_leave_dashboard_extra_hours` to reflect that change

### Note 

- add pending merge hr-attendance manual way

### Results

- before install module
![image](https://github.com/user-attachments/assets/db132a8f-9d8f-4d18-99fd-d73037085711)

- after
![image](https://github.com/user-attachments/assets/ebdddd83-94b6-4d94-bdd2-8a610ea0b3ea)

AaronHForgeFlow committed:
[ADD]hr_attendance_reason v10

AaronHForgeFlow committed:
[ADD] hr_attendance_autoclose (#510)

[UPD] README.rst

[UPD] Update hr_attendance_autoclose.pot

Commit:
[FIX] pylint

Commit:
[MIG][11.0] hr_attendance_reason

Commit:
[MIG][11.0] hr_attendance_autoclose

[ADD] icon.png

[UPD] Update hr_attendance_autoclose.pot

Commit:
[12.0][MIG] hr_attendance_reason

Commit:
[11.0][IMP] float_time widget on hr_attendance_autoclose

hr_attendance_autoclose 12.0.1.1.0

damdam-s committed:
[MIG] hr_attendance_reason: Migration to 13.0

kittiu committed:
[12.0][MIG] hr_attendance_autoclose

[UPD] Update hr_attendance_autoclose.pot

[UPD] README.rst

badbole committed:
Added translation using Weblate (Croatian)

MiquelRForgeFlow committed:
[FIX] compute needs api.depends

hr_attendance_autoclose 12.0.1.0.1

Saran440 committed:
[IMP] hr_attendance_reason: black, isort, prettier

Saran440 committed:
[IMP] hr_attendance_reason: add reason in tree view attendance

[UPD] Update hr_attendance_reason.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-attendance-13.0/hr-attendance-13.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-13-0/hr-attendance-13-0-hr_attendance_reason/

Commit:
[IMP] hr_attendance_autoclose:  black, isort, prettier

Commit:
[13.0][MIG] hr_attendance_autoclose

[UPD] Update hr_attendance_autoclose.pot

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

[UPD] README.rst

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

[UPD] README.rst

dalonsod committed:
[MIG] hr_attendance_reason: Migration to 14.0

[UPD] Update hr_attendance_reason.pot

[UPD] README.rst

aphon61bank committed:
[14.0][FIX] hr_attendance_reason - typo

[UPD] Update hr_attendance_reason.pot

victoralmau committed:
[FIX] hr_attendance_autoclose: Show error in hr.lave form related to employee public (user without employee groups)

[UPD] Update hr_attendance_autoclose.pot

[UPD] README.rst

hr_attendance_autoclose 13.0.1.0.1

victoralmau committed:
[IMP] hr_attendance_autoclose: Change to hr.employee.public

[UPD] Update hr_attendance_autoclose.pot

hr_attendance_autoclose 13.0.1.0.2

victoralmau committed:
[FIX] hr_attendance_autoclose: Show no_autoclose field in view correctly

hr_attendance_autoclose 13.0.1.1.0

victoralmau committed:
Added translation using Weblate (Spanish)

fshah-initos committed:
[IMP] hr_attendance_autoclose: isort, black, prettier.

fshah-initos committed:
[MIG] Migrate module hr_attendance_autoclose to v14.

[UPD] Update hr_attendance_autoclose.pot

[UPD] README.rst

polimex-lgeorgiev committed:
Added translation using Weblate (Bulgarian)

Translated using Weblate (Bulgarian)

Currently translated at 8.6% (2 of 23 strings)

Translation: hr-attendance-14.0/hr-attendance-14.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-14-0/hr-attendance-14-0-hr_attendance_reason/bg/

Translated using Weblate (Bulgarian)

Currently translated at 100.0% (23 of 23 strings)

Translation: hr-attendance-14.0/hr-attendance-14.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-14-0/hr-attendance-14-0-hr_attendance_reason/bg/

petrus-v committed:
[14.0] fix hr attendance autoclose

follow hr.employee instruction any fields on that model must be limited to hr.group_hr_user
to avoid prefetch error while browsing fields

hr_attendance_autoclose 14.0.1.0.1

petrus-v committed:
Added translation using Weblate (French)

Translated using Weblate (French)

Currently translated at 82.6% (19 of 23 strings)

Translation: hr-attendance-14.0/hr-attendance-14.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-14-0/hr-attendance-14-0-hr_attendance_reason/fr/

petrus-v committed:
Added translation using Weblate (French)

brian10048 committed:
[FIX] hr_attendance_autoclose

Employee view did not properly display no_autoclose field

Commit:
[MIG] hr_attendance_reason: Migration to 15.0

[UPD] Update hr_attendance_reason.pot

[UPD] README.rst

HekkiMelody committed:
Added translation using Weblate (Italian)

HekkiMelody committed:
hr_attendance_autoclose: remove almost duplicated field worked_hours from tree

hr_attendance_autoclose 14.0.1.0.2

HekkiMelody committed:
[14.0] [IMP] hr_attendance_autoclose: move autoclose reason to settings and remove hardcoded search

[UPD] Update hr_attendance_autoclose.pot

hr_attendance_autoclose 14.0.1.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-attendance-14.0/hr-attendance-14.0-hr_attendance_autoclose
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-14-0/hr-attendance-14-0-hr_attendance_autoclose/

victoralmau committed:
[MIG] hr_attendance_autoclose: Migration to 15.0

TT40357

victoralmau committed:
[IMP] hr_attendance_reason: Show reasons on attendance screen

TT40897

victoralmau committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (23 of 23 strings)

Translation: hr-attendance-15.0/hr-attendance-15.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-15-0/hr-attendance-15-0-hr_attendance_reason/es/

victoralmau committed:
[IMP] hr_attendance_reason: Add multi-company option + security rule

[UPD] Update hr_attendance_reason.pot

hr_attendance_reason 15.0.1.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-attendance-15.0/hr-attendance-15.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-15-0/hr-attendance-15-0-hr_attendance_reason/

dsolanki-initos committed:
[MIG] hr_attendance_reason: Migration to 16.0

Co-authored-by: dsolanki-initos

TT45376

Commit:
[hr_attendance_overtime_negative_hours] init

HekkiMelody committed:
[MIG] hr_attendance_autoclose: Migration to 16.0

Commit:
[16.0] Add module hr_attendance_calendar_view

OCA-git-bot committed:
[ADD] icon.png

chienandalu committed:
[FIX] hr_attendance_reason: _rpc context overriding

victoralmau committed:
[IMP] hr_attendance_reason: Rename show_reasons_on_attendance_screen field to show_reason_on_attendance_screen

TT40897

mymage committed:
Added translation using Weblate (Italian)

Translated using Weblate (Italian)

Currently translated at 100.0% (23 of 23 strings)

Translation: hr-attendance-15.0/hr-attendance-15.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-15-0/hr-attendance-15-0-hr_attendance_reason/it/

[UPD] Update hr_attendance_reason.pot

[UPD] README.rst

hr_attendance_reason 15.0.1.2.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-attendance-15.0/hr-attendance-15.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-15-0/hr-attendance-15-0-hr_attendance_reason/

victoralmau committed:
[OU-ADD] hr_attendance_reason: Rename res.company field

TT40897

janikvonrotz committed:
update linting config files

albig committed:
Added translation using Weblate (German)

Translated using Weblate (German)

Currently translated at 3.1% (1 of 32 strings)

Translation: hr-attendance-15.0/hr-attendance-15.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-15-0/hr-attendance-15-0-hr_attendance_reason/de/

Translated using Weblate (German)

Currently translated at 90.6% (29 of 32 strings)

Translation: hr-attendance-15.0/hr-attendance-15.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-15-0/hr-attendance-15-0-hr_attendance_reason/de/

danielduqma committed:
[FIX] hr_attendance_reason: search_read params

`search_read` doesn't match definition from core, `read_kwargs` was
missing.

pedrobaeza committed:
[FIX] hr_attendance_calendar_view: Missing __init__.py + README

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-attendance-16.0/hr-attendance-16.0-hr_attendance_autoclose
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-16-0/hr-attendance-16-0-hr_attendance_autoclose/

Commit:
[UPD] Update hr_attendance_calendar_view.pot

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 46.6% (7 of 15 strings)

Translation: hr-attendance-16.0/hr-attendance-16.0-hr_attendance_autoclose
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-16-0/hr-attendance-16-0-hr_attendance_autoclose/es/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: hr-attendance-16.0/hr-attendance-16.0-hr_attendance_calendar_view
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-16-0/hr-attendance-16-0-hr_attendance_calendar_view/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (15 of 15 strings)

Translation: hr-attendance-16.0/hr-attendance-16.0-hr_attendance_autoclose
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-16-0/hr-attendance-16-0-hr_attendance_autoclose/it/

carlos-lopez-tecnativa committed:
[IMP] hr_attendance_reason: pre-commit stuff

xaviedoanhduy committed:
[IMP] hr_attendance_overtime_negative_hours: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] hr_attendance_overtime_negative_hours: Migration to 17.0

xaviedoanhduy commented:
after a look it seems that this module is not needeed anymore: https://github.com/odoo/odoo/pull/108148

lbarry-apsl committed:
[IMP] hr_attendance_calendar_view: black, isort, prettier

nilshamerlinck commented on hr_attendance_overtime_negative_hours/readme/DESCRIPTION.md:
```diff
@@ -0,0 +1 @@
+Override on `hr.leave.type:get_days_all_request()` to also show Negative Hours in Leaves overview (TimeOffCalendar widget)
```
quite technical for a description: just keep `Show Negative Hours in Leaves overview (TimeOffCalendar widget)`

xaviedoanhduy replied:
I have updated it, thanks

xaviedoanhduy commented:
Note:

lbarry-apsl committed:
[MIG] hr_attendance_calendar_view: Migration to 17.0

carlos-lopez-tecnativa committed:
[MIG] hr_attendance_reason: Migration to 17.0

xaviedoanhduy committed:
[MOV] hr_attendance_overtime_nagative_hours: rename to hr_leave_dashboard_extra_time

nilshamerlinck commented on hr_leave_dashboard_extra_hours/static/description/index.html:
this file looks outdated

OCA-git-bot committed:
Merge PR #174 into 17.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update hr_attendance_calendar_view.pot

OCA-git-bot committed:
[BOT] post-merge updates

nguyenminhchien reviewed (approved):
look good but better if adding the unit tests.

OCA-git-bot committed:
Merge PR #178 into 17.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update hr_attendance_reason.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-attendance-17.0/hr-attendance-17.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-17-0/hr-attendance-17-0-hr_attendance_reason/

xaviedoanhduy commented:
I will add tests on c2c side, thank you all for your suggestions.

xaviedoanhduy commented:
Hi @nguyenminhchien, I have added some unit tests

carlos-lopez-tecnativa committed:
[IMP] hr_attendance_autoclose: pre-commit stuff

carlos-lopez-tecnativa committed:
[MIG] hr_attendance_autoclose: Migration to 17.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (38 of 38 strings)

Translation: hr-attendance-17.0/hr-attendance-17.0-hr_attendance_reason
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-17-0/hr-attendance-17-0-hr_attendance_reason/it/

OCA-git-bot committed:
Merge PR #179 into 17.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update hr_attendance_autoclose.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-attendance-17.0/hr-attendance-17.0-hr_attendance_autoclose
Translate-URL: https://translation.odoo-community.org/projects/hr-attendance-17-0/hr-attendance-17-0-hr_attendance_autoclose/
